app/routes/__auth/user.server.ts

```diff
@@ -14,8 +16,59 @@ export async function getUser({ headers }: Request) {
   const email = session.get('email') as string
   const groups = session.get('groups') as string[]
   const idp = session.get('idp') as string | null
-
+  const refreshToken = session.get('refreshToken') as string
+  const cognitoUserName = session.get('cognitoUserName') as string
+  const token = session.get('token') as string
   if (!sub) return null
+  const user = { sub, email, groups, idp, refreshToken, cognitoUserName, token }
+  if (
+    !token ||
+    JSON.parse(Buffer.from(token.split('.')[1], 'base64').toString()).exp *
+      1000 <
+      Date.now()
+  ) {
+    await refreshUser(user)
+  }
+  return user
+}
+
+export async function updateSession(user: any) {
+  const session = await storage.getSession()
+  Object.entries(user).forEach(([key, value]) => {
+    session.set(key, value)
+  })
+  await storage.commitSession(session)
+}
+
+export async function clearUserToken(sub: string) {
```

Comment:
Add a comment to explain what this function does. Something like:

```
// Clear the access tokens from all sessions belonging to the user with the given sub, to force them to get new access tokens.
```

Several of the other functions would benefit from some comments.

Reply:
I added that and one other, are there specific ones you think need comments?